templates/flavors/external-cloud-provider-machinepool/kustomization.yaml

```diff
@@ -0,0 +1,10 @@
+namespace: default
+resources:
+  - ../base
```

Comment:
If you start this from https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/templates/flavors/machinepool it would already have the machinepool and azureclusteridentity (then we can just patch it for flex and OOT)